[clang] Only build clang-tblgen if it is actually needed

It's possible to build clang with an existing clang-tblgen (common when
cross-compiling, for instance) by setting CLANG_TABLEGEN_EXE.  If this
is the case there's no need to build it, as it won't be used.

Upstream-Status: Submitted [https://github.com/llvm/llvm-project/pull/161952]
Signed-off-by: Ross Burton <ross.burton@arm.com>

diff --git i/clang/utils/TableGen/CMakeLists.txt w/clang/utils/TableGen/CMakeLists.txt
index ce759ec8548d..f2294e4389fd 100644
--- i/clang/utils/TableGen/CMakeLists.txt
+++ w/clang/utils/TableGen/CMakeLists.txt
@@ -1,3 +1,4 @@
+if (NOT DEFINED CLANG_TABLEGEN_EXE)
 set(LLVM_LINK_COMPONENTS Support)
 
 add_tablegen(clang-tblgen CLANG
@@ -28,3 +29,4 @@ add_tablegen(clang-tblgen CLANG
   )
 
 target_link_libraries(clang-tblgen PRIVATE clangSupport_tablegen)
+endif()
